File: Mechanical_Behavior_of_Materials_NWU_Shull/python/rotate45matrix.py
import numpy as np

sig = np.zeros((3,3)) # create stress tensor
sig[0][0] = 5e6 # only nonzero components
phi = 45
theta = np.array([[phi,90+phi,90],[90-phi,phi,90],[90,90,0]])
Q = np.cos(np.radians(theta))
QT = np.transpose(Q)
sigp = np.round(Q@sig@QT) #need to use appropiate code to do matrix multiplication in python
print(sigp)

File: Mechanical_Behavior_of_Materials_NWU_Shull/matlab/rotate45matrix.m
%rotate45_matrix.m
sig=zeros(3); % create stress tensor and set to zero
sig(1,1)=5e6; % this is the only nonzero component
phi=45; 
theta=[phi,90-phi,90;90+phi,phi,90;90,90,0];
Q=cosd(theta);
QT=transpose(Q);
sigp=Q*sig*QT

